xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -20,6 +20,7 @@
 import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
+import io.envoyproxy.envoy.api.v2.auth.UpstreamTlsContext;
```

Comment:
We are aiming to not mix envoy's data types into gRPC, so that gRPC's components do not need to import envoy dependencies. In [EnvoyProtoData](https://github.com/grpc/grpc-java/blob/master/xds/src/main/java/io/grpc/xds/EnvoyProtoData.java), we create our own data types corresponding to envoy's and only extract fields needed by gRPC. You may want to consider the same approach to make things more elegant.

Reply:
unrelated note: i was wondering about this when reviewing the SDS. SDS can follow this practice

Reply:
Basically, our thought is to have `XdsClientImpl` being the only interface that interacts with xDS server, do xDS protocol communication and data exchange. Data being used by gRPC is packed in its own format.

Reply:
For now the "downstream code" i.e. the code that receives the ClusterUpdate and then sets the appropriate XdsAttribute expects UpstreamTlsContext so I will need that value. In a future PR I can replace Envoy data types with the ones we define in EnvoyProtoData.

Reply:
Hmm... I hope this could happen at the first place of implementation, as otherwise, people seeing this might follow using envoy's proto types in gRPC directly. Also we should prevent such a spreading going further into gRPC. Since you might have downstream code that already uses `UpstreamTlsContext` proto type directly, you would need a PR to convert that. Could you please put a `TODO` here as a notice to others? Thanks.